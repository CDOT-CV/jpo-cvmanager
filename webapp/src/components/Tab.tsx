import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import { useLocation } from 'react-router-dom'

interface TabProps {
  activeTab: string
  label: string
  path: string

  onClick: (label: string) => void
}

const Tab = (props: TabProps) => {
  const { onClick, path, label } = props
  const location = useLocation()

  let className = 'tab-list-item'

  if (location.pathname.includes(path)) {
    className += ' tab-list-active'
  }

  return (
    <Link
      className={className}
      to={path}
      onKeyDown={(e) => {
        if (e.code === 'Space') {
          onClick(label)
        }
      }}
      onClick={() => onClick(label)}
    >
      {label}
    </Link>
  )
}

Tab.propTypes = {
  activeTab: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
}

export default Tab
