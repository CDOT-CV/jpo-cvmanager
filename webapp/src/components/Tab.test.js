import React from "react";
import { render } from "@testing-library/react";
import Tab from "./Tab";

it("should take a snapshot", () => {
  const { asFragment } = render(<Tab onClick={() => {}} activeTab={""} label={""} />);

  expect(asFragment(<Tab onClick={() => {}} activeTab={""} label={""} />)).toMatchSnapshot();
});
