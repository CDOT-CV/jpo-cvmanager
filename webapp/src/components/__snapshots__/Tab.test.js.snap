// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <li
    class="tab-list-item tab-list-active"
  />
</DocumentFragment>
`;
